ls-ui-react-app/src/appState.ts

```diff
@@ -6,10 +6,10 @@ import {CompilationResult} from "./slices";
 //
 class AppState {
     compilationResult? : CompilationResult
-    errorMsg? : string // An optional error message to show to the user
     compilationIsLoading : boolean = false
-    request? : boolean = false
     apiResponse : any
```

Comment:
As we saw, this `apiResponse` parameter is only being used when there is a non 200-reponse so it is effectively an error case

Reply:
In general the following are mutually exclusive:
 - `compilationResult` -> Success. We have a good compilation result
 - `apiError` and `apiResponse` -> Response is non 200
 - `jsonParseError` -> Reponse was 200 but compilation failed, there will be `errorMsg` parameter, but it's best to check for a failed parse.

Because these are mutually exclusive app state only needs to hold one of these at a time:
```ts

class JsonParseError {
   msg: string
}

class ApiHttpError {
   code: number
   msg: string
}

type ApiResponse = null | JsonParseError | ApiHttpError | CompilationResult;

class AppState {
    compilationIsLoading: boolean = false
    apiResponse: ApiResponse = null
}

// CompilationResult is the usual type from maybe slices.ts?

```